crypto/bio/bio_lib.c

```diff
@@ -856,21 +863,21 @@ int BIO_connect_retry(BIO *bio, int timeout)
 
     if (rv <= 0) {
         if (get_last_sys_error() == ETIMEDOUT) {
-            /*
-             * if blocking, despite blocking BIO, BIO_do_handshake() timed out
-             * when non-blocking, BIO_do_handshake() timed out early
-             * with rv == -1 and get_last_sys_error() == 0
+            /*-
+             * If blocking, BIO_do_handshake() timed out despite blocking BIO.
+             * If non-blocking, BIO_do_handshake() timed out early with
+             *                  rv == -1 and get_last_sys_error() == 0.
              */
             ERR_clear_error();
             (void)BIO_reset(bio);
             /*
-             * unless using BIO_reset(), blocking next connect() may crash and
-             * non-blocking next BIO_do_handshake() will fail
+             * Unless using BIO_reset(), blocking next connect() may crash and
+             * non-blocking next BIO_do_handshake() would fail.
              */
             goto retry;
         } else if (BIO_should_retry(bio)) {
             /* will not actually wait if timeout == 0 (i.e., blocking BIO) */
-            rv = bio_wait(bio, max_time, 100 /* milliseconds */);
+            rv = bio_wait(bio, max_time, 100 /* milliseconds to sleep */);
```

Comment:
I think you mentioned that this is sometimes called for non-blocking bio,
is that right?

Reply:
Yes, this is meant for non-blocking BIOs.
As written in the comment it `bio_wait()` has no effect in the blocking case (timeout == 0).

Reply:
I wonder why a blocking BIO should set BIO_should_retry, that is unexpected.

Reply:
Right, `BIO_should_retry()` should be needed only for the non-blocking case.
I've meanwhile added a respective comment.